keps/sig-network/1880-services-ip-ranges/README.md

```diff
@@ -0,0 +1,1015 @@
+  consume it
+- the configuration is per node, there is no consensus and different apiservers can fight and delete
+  others IPs.
+- the allocation logic is racy, with the possibility of leaking ClusterIPs
+  https://github.com/kubernetes/kubernetes/issues/87603
+- the size of the Service Cluster CIDR, for IPv4 the prefix size is limited to /12, however, for
+  IPv6 it is limited to /112 or fewer. This restriction is causing issues for IPv6 users, since /64
+  is the standard and minimum recommended prefix length
+- only is possible to use reserved a range of Service IP addresses using the feature gate
+  "keps/sig-network/3070-reserved-service-ip-range"
+  https://github.com/kubernetes/kubernetes/issues/95570
+
+### Goals
+
+Implement a new allocation logic for Services IPs that:
+
+- scales well with the number of reservations
+- the number of allocation is tunable
+- is completely backwards compatible
+
+### Non-Goals
+
+- Any change unrelated to Services
+- Any generalization of the API model that can evolve onto something different, like an IPAM API, or
+  collide with existing APIs like Ingress and GatewayAPI.
+- NodePorts use the same allocation model than ClusterIPs, but it is totally out of scope of this
+  KEP. However, this KEP can be a proof that a similar model will work for NodePorts too.
+- Change the default IPFamily used for Services IP allocation, the defaulting depends on the
+  services.spec.IPFamily and services.spec.IPFamilyPolicy, a simple webhook or an admission plugin
+  can set this fields to the desired default, so the allocation logic doesn't have to handle it.
+- Removing the apiserver flags that define the service IP CIDRs, though that may be possible in the future.
+
+## Proposal
+
+The proposal is to implement a new allocator logic that uses 2 new API Objects: ServiceIPRange
+and ServiceIP, and allow users to dynamically increase the number of Services IPs available by
+creating new ServiceIPRanges.
+
+The new allocator will be able to "automagically" consume IPs from any ServiceIPRange available, we can
+think about this model, as the same as adding more disks to a Storage system to increase the
+capacity.
+
+To simplify the model, make it backwards compatible and to avoid that it can evolve into something
+different and collide with other APIs, like Gateway APIs, we are adding the following constraints:
+
+ - a ServiceIPRange will be immutable after creation (to be revisited before Beta).
+ - a ServiceIPRange can only be deleted if there are no Service IP associated to it (enforce by finalizer)
+ - there can be overlapping ServiceIPRanges.
+ - the apiserver will periodically ensure that a "default" ServiceIPRange exists to cover the service CIDR flags
+   and the "kubernetes.default" Service.
+ - any ServiceIP existing in a cluster has to belong to a ServiceIPRange.
+ - any Service with a ClusterIP assigned is expected to have always a ServiceIP object associated.
```

Comment:
Add
- a ServiceIPRange which is being deleted can not be used to allocate new IPs